### ISSUE TITLE
cc65 apple1 target ?

### ISSUE THREAD

**Issue by ijean** [jefftranter/6502#21]
Hi,

I noticed that in one of your Makefiles you are using *cc65* to build for *apple1*, specifically in *https://github.com/jefftranter/6502/tree/master/c/adventure1* however *cc65* on my system errors with:

cl65: No such target system: 'apple1'

Are you using a modified/patched cc65 or the Makefile is wrong ? (I checked the documentation of cc65 and it seems they do not have a backend for apple1)

Thanks,
Jean

**Comment by jefftranter** [comment]
Yes, I used a patched version of cc65 that had an apple1 target.
If interested, the patches are here, but are for an old version of cc65: https://github.com/jefftranter/6502/tree/master/asm/cc65

**Comment by ijean** [comment]
Thanks! I will try to apply your patches to v2.13.3.

**Comment by ijean** [comment]
Your patch still works and I was able to compile a simple hello world program for Apple 1.
It would be great if you have some time to port your patches to a more recent version of cc65. (It is a bit over my knowledge level now to even attempt to do the port myself).
Thanks for all your hard work in supporting 6502 fans!

**Comment by ijean** [comment]
Sorry to bother you with another question. I build a simple hello world program using:

cl65 -O -t apple1 hello.c

However, I have no idea at what memory address I need to load the resulting code. I tried (using the Woz monitor) addresses 0100, 0200 without success (I use an emulator called POM1 to test). Can you please let me know at which memory address am I supposed to load the program ? 

For running the program, I assume it is the regular xxxxR (where xxxx is the address at which the program was loaded).

Thanks

**Comment by jefftranter** [comment]
The start address should be 0280.
If you generate a map file when you build it with cc65, it will list the address of the STARTUP section.

The output of cc65 is a binary file.
I wrote a program called bintomon that will convert a binary to the Woz monitor format. You can then send this file to the Apple 1 over the serial port (I'm not sure how you would do it with an emulator).
See https://github.com/jefftranter/6502/tree/master/util/bintomon

This blog post I wrote might also be helpful:
http://jefftranter.blogspot.com/2012/04/c-programming-tutorial-with-cc65-on.html

**Comment by ijean** [comment]
Thanks!

Actually I read your blog post earlier today. (If it is not too much trouble, I suggest to add the link to your blog post in the README of the cc65 folder from GitHub, this may be helpful for other people too.)

Turns out the POM1 emulator has a slightly different file format for memory files than the one for replica1, but I was able to adapt your bintomon.c file for the POM1 emulator and now I can load and run compiled programs!
